manifests/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -1,10 +1,14 @@
+{{ if or (eq .Values.revision "") (not .Values.clusterResources) }}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: istiod
+  name: istiod{{ .Values.revision }}
   namespace: {{ .Release.Namespace }}
   labels:
     app: pilot
+    {{- if ne .Values.revision ""}}
+    version: {{ .Values.revision }}
```

Comment:
should be `istio.io/rev`?

Reply:
It's a bit more complicated - for deployments we already document (or at least use in examples) that 'version' is special for telemetry. 

@mandarjog 

We can fix in separate PR - but we need to sync with dashboard and other telemetry-related things.

Reply:
I don't think this applies here though. We can add both, maybe. But the user pods will definitely have the `istio.io/rev` label, not `version`, since we don't want to conflict with their labels.

Reply:
The pods may have istio.io/rev to indicate which CP they are using, and version to indicate their own version.

For istiod - 'version' represents the version of istiod - from telemetry and service selection perspective. 
We can add a istio.io/rev - but not sure what would be the semantic, istiod is not using a CP.

Telemetry will need to track both - but I think it should have clear semantics:
- version means the version of the application itself ( v1, v2, appcanary ) - which in case of istiod is the revision.
- istio.io/rev means what control plane is used by the app.

Reply:
That works for me